### PR TITLE
Changelog 2.0

### PR DESCRIPTION
**Pull request by hannesveit**
A preliminary changelog for the 2.0 release including a tab-separated table of parameter changes together with a small script to generate it.

As I've been rather inactive lately, I may not be aware of some of the important highlights of this release that need to be mentioned explicitly. Feel free to add more...

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -8,18 +8,638 @@ INIFileEditor - graphical parameter editor for INI files
...
-Release date: unknown
+Release date: January 2015
 
-OpenMS 1.12.0 is the first release after the switch to git and a complete overhaul of the
-build system. Furthermore we removed the dependency to GSL in OpenMS 1.12.0 and replaced
-the functionality with Eigen3 and Wildmagic.
+OpenMS 2.0 is the first release after the switch to git and a complete overhaul of the
+build system. It introduces a considerable number of new features and bug fixes.
+Furthermore, we removed the dependency to GSL and replaced the functionality using
+Eigen3 and Wildmagic.
```
Comment (hroest) [inline CHANGELOG]: state that means that now the core and the full build is non-GPL (e.g. either Apache or BSD) licence

File: CHANGELOG [hunk 2]
```diff
@@ -8,18 +8,638 @@ INIFileEditor - graphical parameter editor for INI files
...
+
+File formats:
+  - mzQuantML support (experimental)
+  - mzIdentML support (experimental)
+  - mzTab support (experimental)
```
Comment (hroest) [inline CHANGELOG]: indexedmzML as well

File: CHANGELOG [hunk 3]
```diff
@@ -8,18 +8,638 @@ INIFileEditor - graphical parameter editor for INI files
...
+Third party software:
+  - Added Fido support
+  - Added MS-GF+ support
+
+Changes to the Build System / Package System:
+  - Restructured repository layout
+  - Added support for Travis CI
+  - Simplified pyOpenMS build system
+
+Changed parameters:
```
Comment (hroest) [inline CHANGELOG]: can we have this in a separate file? it will be unreadable very fast otherwise ...

**Comment by hroest** [comment]
these are set of improvements that I wrote down and would be nice in the changelog

**Comment by hroest** [comment]
these are set of improvements that I wrote down and would be nice in the changelog

- Low Memory Algorithms
- Indexed mzML
- numpress encoding schema
- OpenSWATH single binary (high performance integrated workflow)
- single interface to multiple different implementations to ways to
access mass spectrometric data  (in memory, on disk with index, cached
on disc for fast access) plus a chainable, low memory sequential
processor of MS data (using a separate interface)
- FeatureFinderIdentification
- RNA xlink
- Metabolite Spectral Matcher
- QcMLHandler
- GSL removal
- AASequence refactoring
- VS2013 support
- significant improvements in MzML parsing
(https://github.com/OpenMS/OpenMS/pull/702)
- completely reworked directory structure and build system
- speedup of parsing (String -> double conversions)
- Python 3.x support

**Comment by cbielow** [comment]
I like it.

Maybe we can remove commits with just
'fix spellcheck[s]'
or
'fix compile error'
Not that interesting for the user IMHO.


Am 1/15/2015 um 4:52 PM schrieb Johannes Veit:

**Comment by hannesveit** [comment]
Thanks for your comments.

I have added most of your suggestions, @hroest. But I think I would prefer to leave the changed parameters table in the CHANGELOG rather than adding another file for that purpose. IMO, it doesn't hurt much as it is at the end of the section for release 2.0 (together with the full list of resolved issues / merged pull requests, which is also not *that* interesting). You don't have to read in this area unless you really want to ;)

@cbielow: You're right, there's some rather uninformative stuff in there... I didn't want to manually select a few issues / pull requests that are "not interesting enough". So I tried to find a systematic way to shorten that list. But unfortunately, by removing, e.g., all PRs with the [NOP] tag, we also lose interesting stuff, e.g. "[NOP] add heavy methyl SILAC labels to unimod.xml" etc. So I left it as it was. I think it's OK to keep it for the sake of completeness as it doesn't hurt much.

Idea: maybe we should introduce yet another tag, something like [IGNORE] or so, for stuff that isn't worth mentioning in the CHANGELOG. That would make it easier to generate a decent list of merged pull requests for future releases.

**Comment by cbielow** [comment]
> systematic way to shorten that list. But unfortunately, by removing,
 > e.g., all PRs with the [NOP] tag, we also lose interesting stuff,
 > e.g. "[NOP] add heavy methyl SILAC labels to unimod.xml" etc.

sounds like a clear mis-use of NOP to me.
NOP should be whitespace and formatting changes... thats it.

**Comment by timosachsenberg** [comment]
+1 I agree. Maybe we need to take a closer look for correct pull request names in the future.

**Pull request merged**